pkg/splunk/enterprise/util.go

```diff
@@ -32,6 +34,54 @@ import (
 // kubernetes logger used by splunk.enterprise package
 var log = logf.Log.WithName("splunk.enterprise")
 
+// regex to extract the region from the s3 endpoint
+var regionRegex = ".*.s3[-,.](?P<region>.*).amazonaws.com"
+
+//getRegion extracts the region from the endpoint field
+func getRegion(endpoint string) string {
+	pattern := regexp.MustCompile(regionRegex)
+	return pattern.FindStringSubmatch(endpoint)[1]
+}
+
+// GetRemoteStorageClient returns the corresponding S3Client
+func GetRemoteStorageClient(client splcommon.ControllerClient, cr splcommon.MetaObject, appFrameworkRef *enterprisev1.AppFrameworkSpec, vol *enterprisev1.VolumeSpec, location string) (splclient.S3Client, error) {
+
+	scopedLog := log.WithName("GetRemoteStorageClient").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	//use the provider name to get the corresponding function pointer
+	getClient, _ := splclient.S3Clients[vol.Provider]
+
+	appSecretRef := vol.SecretRef
+	s3ClientSecret, err := splutil.GetSecretByName(client, cr, appSecretRef)
+	if err != nil {
+		return nil, err
```

Comment:
Nit: seems like a generic error. Probably, the best place to add an error is in GetSecretByName itself, I don't know why we didn't do it before :-).

Reply:
Ok, i'll add an error log in GetSecretByName